ddtrace/contrib/requests/connection.py

```diff
@@ -17,23 +18,10 @@
 
 def _extract_hostname(uri):
     # type: (str) -> str
-    end = len(uri)
-    j = uri.rfind("#", 0, end)
-    if j != -1:
-        end = j
-    j = uri.rfind("&", 0, end)
-    if j != -1:
-        end = j
-
-    start = uri.find("://", 0, end) + 3
-    i = uri.find("@", start, end) + 1
-    if i != 0:
-        start = i
-    j = uri.find("/", start, end)
-    if j != -1:
-        end = j
-
-    return uri[start:end]
+    parsed_uri = parse.urlparse(uri)
+    if parsed_uri.port:
```

Comment:
The requests framework test [fails](https://github.com/DataDog/dd-trace-py/runs/7755898156?check_suite_focus=true) on this line. If parsed_uri.port is an invalid value then a [value error is raised](https://github.com/python/cpython/blob/main/Lib/urllib/parse.py#L175). We should wrap this code block in a try/except.